go/private/repositories.bzl

```diff
@@ -92,5 +92,5 @@ def go_repositories(
 
   bzl_format_repositories()
 
-  go_repository_select(go_version, go_linux, go_darwin)
+  go_repository_select(name = "io_bazel_rules_go_toolchain", go_version = go_version)
```

Comment:
I decided not to change the signature of the function right now, instead I think we should probably add deprecated warnings if we see people using it, and only break the signature when we fully switch to toolchains and drop all pretense at using this method to select the toolchain (the version arg will also dissapear)